airflow/utils/log/log_reader.py

```diff
@@ -40,7 +40,7 @@ class TaskLogReader:
     """Time to sleep between loops while waiting for more logs"""
 
     def read_log_chunks(
-        self, ti: TaskInstance, try_number: int | None, metadata
+        self, ti: TaskInstance, try_number: int | None, metadata, page_number: int | None = None
```

Comment:
do we really need a new page_number param?  doesn't metadata have support for an offset?

Reply:
I'll look into that

Reply:
@andyjianzhou did you get a chance to look into the metadata param?

Reply:
I was thinking that in general we should use the offset/limit approach rather than maintaining a page number. We can (should?) still show the page number in the web UI, but those should always be converted in the frontend to an offset to talk to the log server.

Reply:
I have found these two key variables in the metadata, `log_pos`, and `offset` that can help with pagination. I agree that switching to `log_pos` and `offset` for pagination would simplify the approach. I'll proceed with updating the `read_log_chunks` method to support `log_pos` and modify the `get_log` endpoint to use these for pagination instead. Currently, we are using the page_number parameter to read log chunks, but I'll create tests to ensure that the changes I'm making isn't breaking any other functionality

Reply:
So I have a question, you brought up the usage of using offset/limit approach, however wouldn't it be a lot more simpler and intuitive if we used `log_pos`?